### PR TITLE
Add generateErrorLog setting to Run-ALPipeline

### PR DESCRIPTION
**Pull request by mazhelez**
Add generateErrorLog setting to `Run-ALPipeline`

The setting is propagated down to `Compile-AppInNavContainer` and `Compile-AppWithBcCompilerFolder` and is used to set the compiler setting `errorLog`.

The generated error log file has the same name as the `.app` file but the file extension is `.errorLog.json`.

### CODE REVIEW

File: AppHandling/Run-AlPipeline.ps1 [hunk 1]
```diff
@@ -13,6 +13,8 @@
   License file to use for AL Pipeline.
  .Parameter containerName
   This is the containerName going to be used for the build/test container. If not specified, the container name will be the pipeline name followed by -bld.
+ .Parameter generateErrorLog
+  Path to file where to output the alerts from the compiler.
```
Comment (aholstrup1) [inline AppHandling/Run-AlPipeline.ps1:17]:
```suggestion
  Include this switch if compiler should output the errorlog as a file
```

File: AppHandling/Run-AlPipeline.ps1 [hunk 2]
```diff
@@ -2426,6 +2431,28 @@ if (!(Test-Path $destFolder -PathType Container)) {
...
+# Copy error logs to build artifact folder
+if($generateErrorLog) {
+    # Define destination folder for error logs
+    $destFolder = Join-Path $buildArtifactFolder "ErrorLogs"
+    if (!(Test-Path $destFolder -PathType Container)) {
+        New-Item $destFolder -ItemType Directory | Out-Null
+    }
+
+    # Copy error logs for all apps
+    @($apps, $testApps, $bcptTestApps) | ForEach-Object {
```
Comment (aholstrup1) [inline AppHandling/Run-AlPipeline.ps1:2444]: Is it only apps that successfully compiled that are in $apps, $testapps and $bcptTestApps? If an app fails to compile shouldn't we still get the errorlog?

Comment (mazhelez) [reply AppHandling/Run-AlPipeline.ps1:2444]: That's a good point, but I'm not sure the log file is generated if the app fails compiling. I'll check.

File: CompilerFolderHandling/Compile-AppWithBcCompilerFolder.ps1 [hunk 3]
```diff
@@ -382,6 +383,11 @@ try {
         $alcParameters += @("/nowarn:$nowarn")
     }
 
+    if ($generateErrorLog) {
+        $errorLogFilePath = $appOutputFile -replace ".app$", ".errorLog.json"
```
Comment (freddydk) [inline CompilerFolderHandling/Compile-AppWithBcCompilerFolder.ps1:387]: Single quotes? (when a $ sign is present)

File: AppHandling/Run-AlPipeline.ps1 [hunk 4]
```diff
@@ -13,6 +13,8 @@
   License file to use for AL Pipeline.
  .Parameter containerName
   This is the containerName going to be used for the build/test container. If not specified, the container name will be the pipeline name followed by -bld.
+ .Parameter generateErrorLog
+  Switch parameter on whether to generate an alerts log file. Default is false.
```
Comment (freddydk) [inline AppHandling/Run-AlPipeline.ps1:19]: Add to the comment that an alerts log file is generated for each compiled app file and placed in the output folder next to the app file.
(I think that is what it does right?)

**Review by freddydk — CHANGES_REQUESTED** [review]

File: AppHandling/Run-AlPipeline.ps1 (hunk 2 above)
Comment (mazhelez) [reply AppHandling/Run-AlPipeline.ps1]: Done, now the file are copied always.

**Review by freddydk — APPROVED** [review]

**Pull request merged**